aws/aws-sdk-js-v3
# feat(clients): change runtime config from constant to provider functions

### Issue
Resolves #2557 

### Description
This change makes runtime default configs as a function instead of constant. As a result, different client instance won't share the same runtime config (e.g. HTTP handlers). This change also uses inline nullish coalescing in replace of fully loaded default configuration. This change makes sure the SDK won't load default value/provider at all when corresponding custom config is supplied. 

This change may increase the code size because each nullish coalescing operator is transcribed to more inline code. However the difference is unnoticeable, tested on a React project with only S3 and GetObject command. 

### Testing
* unit test
* manual test(S3, ACM)
* API reference 

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
